### PR TITLE
Contracts tests time optimization

### PR DESCRIPTION
**Pull request by szotov**
* Reset state for chain each module
* Stub for gas estimation while `transact`

### CODE REVIEW

File: tests/fixtures.py [hunk 1]
```diff
@@ -72,6 +74,7 @@
     MOCK_REGISTRY_FILEPATH,
     TEST_GAS_LIMIT,
     INSECURE_DEPLOYMENT_SECRET_HASH,
+    NUMBER_OF_ETH_TEST_ACCOUNTS,
```
Comment (mswilkison) [inline tests/fixtures.py]: I think you can remove this since you reverted the line where it's used in 800876

Comment (szotov) [reply tests/fixtures.py]: ✔️

File: tests/fixtures.py [hunk 2]
```diff
@@ -366,6 +368,14 @@ def _make_testerchain():
     eth._utils.headers.GAS_LIMIT_MINIMUM = TEST_GAS_LIMIT
     eth._utils.headers.GENESIS_GAS_LIMIT = TEST_GAS_LIMIT
     eth.vm.forks.frontier.headers.GENESIS_GAS_LIMIT = TEST_GAS_LIMIT
+
+    # Monkey patch to prevent gas estimates
+    def _get_buffered_gas_estimate(web3, transaction, gas_buffer=100000):
+        return TEST_GAS_LIMIT
+
+    import web3
+    web3.eth.get_buffered_gas_estimate = _get_buffered_gas_estimate
```
Comment (KPrasch) [inline tests/fixtures.py]: How does this differ from setting a Web3 gas strategy and why not use that API?

Comment (szotov) [reply tests/fixtures.py]: It's related to gas estimations itself, gas price is independent from that

**Pull request merged**